doc/source/security/encryption-functions.rst

```diff
@@ -0,0 +1,431 @@
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+   * - default
+     - 10000
+
+The range for this variable is from 1024 to 10000.
+
+.. _dsa-threshold:
+
+.. rubric:: `encryption_udf.dsa_bits_threshold`
+
+These variable sets the threshold limits for :ref:`asymmetric_encrypt <asym-encrypt>`. The value takes precedence over the OpenSSL maximum length value.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+
+
+The range for this variable is from 1024 to 9984.
+
+.. _rsa-threshold:
+
+.. rubric:: `encryption_udf.rsa_bit_threshold`
+
+These variable sets the threshold limits for the :ref:`asymmetric_encrypt <asym-encrypt>`. The value takes precedence over the OpenSSL maximum length value.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+   * - default
+     - 16384
+
+The range for this variable is from 1024 to 16384.
+
+.. _function-examples:
+
+.. rubric:: Examples
+
+The following example sets the threshold variables, creates a private key, creates a public key, encrypts data, and decrypts data.
+
+
+.. code-block:: mysql
+
+    -- Set Global variable
+    mysql> SET GLOBAL encryption_udf.dh_bits_threshold = 4096;
+
+.. code-block:: mysql
+
+    -- Create private key
+    mysql> SET @private_key = create_asymmetric_priv_key('RSA', 3072);
+
+    -- Create public key
+    mysql> SET @public_key = create_asymmetric_pub_key('RSA', @private_key);
+
+    -- Encrypt data using the private key (you can also use the public key)
+    mysql> SET @ciphertext = asymmetric_encrypt('RSA', 'This text is secret', @private_key);
+
+    -- Decrypt data using the public key (you can also use the private key)
+    -- The decrypted value @plaintext should be identical to the original 'This text is secret'
+    mysql> SET @plaintext = asymmetric_decrypt('RSA', @ciphertext, @public_key);
+
+The following example generates a digest, generates a digest signature, and verifies the signature against the digest:
+
+.. code-block:: mysql
+
+   -- Generate a digest string
+   mysql> SET @digest = create_digest('SHA256', 'This is the text for digest');
+
+   -- Generate a digest signature
+   mysql> SET @signature = asymmetric_sign('RSA', @digest, @private_key, 'SHA256');
+
+   -- Verify the signature against the digest
+   -- The @verify_signature must be equal to 1
+   mysql> SET @verify_signature = asymmetric_verify('RSA', @digest, @signature, @public_key, 'SHA256');
+
+The following examples create a symmetric keys using Diffie-Hellman parameters.
```

Comment:
The following example creates symmetric keys using Diffie-Hellman parameters.

Reply:
Updated.